### ISSUE TITLE
0.7.0 release doesn't store PGP keys in keychain in my iPhone?

### ISSUE THREAD

**Issue by rayx** [mssun/passforios#299]
Hi, I installed Pass for iOS from Apple App Store in China Mainland. The version shown in its `Setting->About` is `Pass for iOS 0.7.0(15)`. I chose to save all the credential information locally when I configured the application, which included:

- Git repo's username and password (I access the repo through https)
- PGP keys and the passphrase

Everything worked fine and I added a few passwords. Then I checked the phone's keychain but only saw the git repo's user name and password. There was no PGP private key or its passphrase in keychain.

I'm not familiar with iOS programming, but from the discussion in issue https://github.com/mssun/passforios/issues/223, my impression is that with PR https://github.com/mssun/passforios/pull/278 the app should stores PGP key in keychain. Git commit history shows that 0.7.0 release should include that commit. So I wonder why that's not the case with my phone?

Thanks for any help.

**Comment by yishilin14** [comment]
你好。

0.7.0 版本里，SSH key，PGP key 和 Git 账户密码都是存在 Keychain 里的。所有 Keychain
的访问都是通过这个类完成的：
https://github.com/mssun/passforios/blob/master/passKit/Helpers/AppKeychain.swift

想问一下你是如何查看手机的 keychain 的？

Huan Xiong <notifications@github.com> 于2019年8月29日周四 下午8:55写道：

**Comment by rayx** [comment]
@yishilin14 I look at the contents of keychain by selecting `Accounts & Passwords->App & Website Passwords' in `Settings` app. I think the saved PGP key should be in it, isn't it?

Thanks for the pointer. It seems that the code uses a 3rd party swift wrapper to access keychain. Unfortunately I don't do iOS programming so I can't build and debug code to investigate it. On the other hand, however, the app does work fine on my phone so I have no idea what goes wrong. Could I misunderstand something?

**Comment by rayx** [comment]
> @yishilin14 I look at the contents of keychain by selecting `Accounts & Passwords->App & Website Passwords` in `Settings` app. I think the saved PGP key should be in it, isn't it?

I think I figured out why. While iOS has only a single keychain, an app can only access the items it saved in the keychain. Since the Settings app is signed by Apple and Pass app is signed by the project's author, it's impossible for Settings app to get the items saved by Pass app. The passwords shown in Settings apps perhaps came from Safari app which is also signed by Apple. 

So I did an experiment to verify the above understanding. I deleted the app and reinstalled it. After it started I checked it settings. The PGP key and the app's passcode was still there (perhaps because deleting an app didn't delete its items in keychain), but the git repo's setting was gone. Just curious, are you sure git repo's name and password are also stored in keychain? It's a bit hard to tell this from the experiment result because repo setting includes other information like repo URL and branch and the code perhaps requires all information are available.
